src/main/java/core/basesyntax/utils/generate/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.utils.generate.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.generate.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final ReportGenerator generator;
+
+    public ReportGeneratorImpl(ReportGenerator generator) {
+        this.generator = generator;
+    }
+
+    @Override
+    public String getReport(StorageImpl storage) {
+        return generator.getReport(storage);
+    }
+}
```

Comment:
This implementation of `ReportGeneratorImpl` is not following the SOLID principles. It seems to be a wrapper that doesn't add any functionality to the `ReportGenerator` it receives. Please ensure that each class has a single responsibility and provides a unique functionality.